File: baekjoon/DFS&BFS/16953_AtoB.py
from collections import deque
import sys
input = sys.stdin.readline
n, m = map(int, input().split())

def solution():
    queue = deque([[n, 1]])

    while queue:
        [number, cnt] = queue.popleft()

        if number == m:
            return cnt

        if number * 2 <= m:
            queue.append([number * 2, cnt + 1])
        if int(str(number) + '1') <= m:
            queue.append([int(str(number) + '1'), cnt + 1])

    return -1


print(solution())

File: baekjoon/DFS&BFS/16953_AtoB.js
const fs = require('fs')
const nm = fs.readFileSync('../input.txt').toString().trim()
const [n, m] = nm.split(' ').map(Number)

function solution() {
  const queue = [[n, 1]]
  while (queue.length) {
    const [number, cnt] = queue.shift()

    if (number === m) {
      return cnt
    }

    if (number * 2 <= m) queue.push([number * 2, cnt + 1])
    if (Number(number + '1') <= m) queue.push([Number(number + '1'), cnt + 1])
  }

  return -1
}

console.log(solution())
